magic-gds-drc/run_all_drc.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Copyright 2020 Efabless Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+-------------------------------------------------------------------------
+run_all_drc.py --- A script that will run run_standard_drc
+for all .gds files under the cells/ folder. Must be run from repository
+root.
+
+Usage: python3 run_all_drc.py --help
+
+Results:
+
+  Prints a report to standard output.
+	
+-------------------------------------------------------------------------
+"""
+
+import os
+import re
+import click
+import traceback
+import subprocess
+from typing import List, Tuple
+import concurrent.futures as fut
+
+acceptable_errors_list = []
+
+SCRIPT_DIR = os.path.realpath(os.path.dirname(__file__))
+STANDARD_DRC_SCRIPT = os.path.join(SCRIPT_DIR, "run_standard_drc.py")
+PDK_SUBSET = os.path.join(SCRIPT_DIR, "sky130A")
+
+DRCError = Tuple[str, List[str]]
+
+def parse_drc_report(report: str) -> List[DRCError]:
+    """
+    Takes a magic report in the format:
+
+    Header\\n\\n
+    Error 1\\nError\\nDetails\\n\\n
+    Error 2\\nError\\nDetails\\n\\n
+    [...]
+    Error N\\nError\\nDetails\\n\\n
+
+    and returns all errors as a list of tuples, where the first element of the
+    tuple is the name of the error and the other lines are the details.
+    """
+    components = list(map(lambda x: x.split("\n"), report.split("\n\n")))
+    
+    errors = []
+    for error in components[1:]: # Header is components[0]
+        error_name = error[0]
+        if error_name in acceptable_errors_list:
+            continue
+        errors.append((error[0], error[1:]))
+    
+    return errors
+
+def drc_gds(path: str) -> Tuple[str, List[DRCError]]:
+    """
+    Takes a GDS path. Returns the name of the cell and returns a list of
+    DRC errors.
+    """
+    cell_name = os.path.basename(path)[:-4]
+    
+    env = os.environ.copy()
+    env["PDKPATH"] = PDK_SUBSET
+
+    res = subprocess.run([
+        "python3",
+        STANDARD_DRC_SCRIPT,
+        path
+    ], env=env, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    
+    report_path = path[:-4] + "_drc.txt"
+    try:
+        report = open(report_path).read()
```

Comment:
If debugging is turned on, it would probably be good to dump the whole `_drc.txt` file into a group.
```python

# Enable extra debug output when ACTIONS_STEP_DEBUG=true in the environment.
# https://docs.github.com/en/actions/managing-workflow-runs/enabling-debug-logging#enabling-step-debug-logging
if os.environ.get('ACTIONS_STEP_DEBUG', 'false') == 'true':
  def debug_dump(title, data):
    print("::group::"+title)
    print(data)
    print("::endgroup::")
else:
  def debug_dump(title, data):
    pass